src/ucs/datastruct/ptr_map.inl

```diff
@@ -79,6 +79,7 @@ ucs_ptr_map_put(ucs_ptr_map_t *map, void *ptr, int indirect,
     if (ucs_likely(!indirect)) {
         *key = (uintptr_t)ptr;
         ucs_assert(!(*key & UCS_PTR_MAP_KEY_MIN_ALIGN));
+        ucs_assert(*key != 0);
```

Comment:
why needed?
maybe assert in func start that ptr != NULL?

Reply:
@yosefe due to the discussion here - https://github.com/openucx/ucx/pull/5769/files#r503383561